primedev/scripts/client/scriptmodmenu.cpp

```diff
@@ -1,178 +1,153 @@
+			continue;
 		}
-	}
 
-	return SQRESULT_NULL;
-}
+		g_pSquirrel<context>->pushnewstructinstance(sqvm, 9);
 
-ADD_SQFUNC("string", NSGetModVersionByModName, "string modName", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
-{
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
+		// name
+		g_pSquirrel<context>->pushstring(sqvm, mod.Name.c_str(), -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 0);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
-	{
-		if (!mod.Name.compare(modName))
-		{
-			g_pSquirrel<context>->pushstring(sqvm, mod.Version.c_str());
-			return SQRESULT_NOTNULL;
-		}
-	}
+		// description
+		g_pSquirrel<context>->pushstring(sqvm, mod.Description.c_str(), -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 1);
 
-	return SQRESULT_NULL;
-}
+		// version
+		g_pSquirrel<context>->pushstring(sqvm, mod.Version.c_str(), -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 2);
 
-ADD_SQFUNC("string", NSGetModDownloadLinkByModName, "string modName", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
-{
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
+		// download link
+		g_pSquirrel<context>->pushstring(sqvm, mod.DownloadLink.c_str(), -1);
+		g_pSquirrel<context>->sealstructslot(sqvm, 3);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
-	{
-		if (!mod.Name.compare(modName))
-		{
-			g_pSquirrel<context>->pushstring(sqvm, mod.DownloadLink.c_str());
-			return SQRESULT_NOTNULL;
-		}
-	}
+		// load priority
+		g_pSquirrel<context>->pushinteger(sqvm, mod.LoadPriority);
+		g_pSquirrel<context>->sealstructslot(sqvm, 4);
 
-	return SQRESULT_NULL;
-}
+		// enabled
+		g_pSquirrel<context>->pushbool(sqvm, mod.m_bEnabled);
+		g_pSquirrel<context>->sealstructslot(sqvm, 5);
 
-ADD_SQFUNC("int", NSGetModLoadPriority, "string modName", "", ScriptContext::SERVER | ScriptContext::CLIENT | ScriptContext::UI)
-{
-	const SQChar* modName = g_pSquirrel<context>->getstring(sqvm, 1);
+		// required on client
+		g_pSquirrel<context>->pushbool(sqvm, mod.RequiredOnClient);
+		g_pSquirrel<context>->sealstructslot(sqvm, 6);
 
-	// manual lookup, not super performant but eh not a big deal
-	for (Mod& mod : g_pModManager->m_LoadedMods)
-	{
-		if (!mod.Name.compare(modName))
+		// is remote
+		g_pSquirrel<context>->pushbool(sqvm, mod.m_bIsRemote);
+		g_pSquirrel<context>->sealstructslot(sqvm, 7);
+
+		// convars
+		g_pSquirrel<context>->newarray(sqvm);
+		for (ModConVar* cvar : mod.ConVars)
 		{
-			g_pSquirrel<context>->pushinteger(sqvm, mod.LoadPriority);
-			return SQRESULT_NOTNULL;
+			g_pSquirrel<context>->pushstring(sqvm, cvar->Name.c_str());
+			g_pSquirrel<context>->arrayappend(sqvm, -2);
 		}
+		g_pSquirrel<context>->sealstructslot(sqvm, 8);
+
+		// add current object to squirrel array
+		g_pSquirrel<context>->arrayappend(sqvm, -2);
```

Comment:
since this is duplicated code why not make a function like `void modToSquirrel(HSquirrelVm* sqvm,Mod& mod)` so when you change the struct you have one less place to change

Reply:
Implemented in 9bda30cff38564a38a09023f841bba25819e268d.